### PR TITLE
Added IntelOwlClient struct with some methods.

### PR DESCRIPTION
**Pull request by spiderxm**

### CODE REVIEW

File: gointelowl/gointelowl.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Struct for IntelOwl.
+// Initiate it using Token, Instance, Certificate
+type IntelOwl struct {
+	Token       string
+	Instance    string
```
Comment (eshaan7) [inline gointelowl/gointelowl.go]: Rename `Instance` to `URL`

File: gointelowl/gointelowl.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	Certificate string
+}
+
+// Fetch list of all tags.
+//        Endpoint: ``/api/tags``
+// It Returns Slice of Tags
+func (intelOwl IntelOwl) GetAllTags() []map[string]string {
```
Comment (Eshaan7) [inline gointelowl/gointelowl.go:21]: this looks little confusing to me, can you use `client IntelOwl` as the variable name instead ?

Comment (Eshaan7) [inline gointelowl/gointelowl.go:21]: this looks little confusing to me, can you use `ioClient IntelOwlClient` as the variable name instead ?

File: gointelowl/gointelowl.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+// Struct for IntelOwl.
+// Initiate it using Token, Instance, Certificate
+type IntelOwl struct {
```
Comment (eshaan7) [inline gointelowl/gointelowl.go]: should rename this to `IntelOwlClient` for clarity.

File: gointelowl/gointelowl.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	Instance    string
+	Certificate string
+}
+
+// Fetch list of all tags.
+//        Endpoint: ``/api/tags``
+// It Returns Slice of Tags
+func (intelOwl IntelOwl) GetAllTags() []map[string]string {
+	url := intelOwl.Instance + "/api/tags"
+	request, err := http.NewRequest("GET", url, nil)
```
Comment (eshaan7) [inline gointelowl/gointelowl.go]: Line 23-34 are similar in all these methods, so we should extract this into an utility function like `buildAndMakeRequest`.

File: gointelowl/gointelowl.go (hunk 2 above)
Comment (eshaan7) [inline gointelowl/gointelowl.go]: can you use `client IntelOwlClient` as the variable name instead ?

**Comment by eshaan7** [comment]
Very good work! I have just requested small naming changes here and there.

**Comment by spiderxm** [comment]
Ok, I'll make the changes.

File: gointelowl/gointelowl.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		log.Fatalln(err)
+	}
+	log.Println(url, response.StatusCode)
+	return response
+}
+
+// Fetch list of all tags.
+//        Endpoint: ``/api/tags``
+// It Returns Slice of Tags
+func (client IntelOwlClient) GetAllTags() []map[string]string {
```
Comment (eshaan7) [inline gointelowl/gointelowl.go]: we should pass the pointer to the client and not the instance so this should be `client *IntelOwlClient`.

Comment (eshaan7) [reply gointelowl/gointelowl.go]: same change applies to other methods

File: gointelowl/gointelowl.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+// Struct for IntelOwl.
+// Initiate it using Token, Instance, Certificate
+type IntelOwlClient struct {
+	Token       string
+	URL         string
+	Certificate string
+}
+
+// Take url and IntelOwlClient and place request and return response
+func buildAndMakeGetRequest(url string, client IntelOwlClient) *http.Response {
```
Comment (eshaan7) [inline gointelowl/gointelowl.go]: pass the token directly and not the client object

**Comment by eshaan7** [comment]
Also, move `gointelowl.go` to the root, no need for `gointelowl` folder.

**Comment by eshaan7** [comment]
At last. please don't forget to squash all commits into one!

**Comment by spiderxm** [comment]
ok, sure

**Pull request merged**